src/core/components/courses/TraineeTable.vue

```diff
@@ -22,56 +22,21 @@
         </ni-responsive-table>
         <q-card-actions align="right" v-if="canEdit">
           <q-btn no-caps flat color="primary" icon="add" label="Ajouter un stagiaire" :disable="loading"
-            @click="additionModal = true" />
+            @click="traineeCreationModal = true" />
         </q-card-actions>
       </q-card>
     </div>
 
     <!-- Add trainee modal -->
-    <ni-modal v-model="additionModal" @hide="resetAddTraineeForm">
-      <template slot="title">
-        Ajouter un <span class="text-weight-bold">stagiaire</span> à la formation
-      </template>
-      <ni-input :disable="!firstStep" in-modal v-model.trim="newTrainee.local.email" required-field :last="firstStep"
-        @blur="$v.newTrainee.local.email.$touch" caption="Email" :error-message="emailError($v.newTrainee)"
-        :error="$v.newTrainee.local.email.$error" />
-      <template v-if="!firstStep">
-        <ni-input in-modal v-if="!addNewTraineeCompanyStep" v-model="newTrainee.identity.firstname" caption="Prénom" />
-        <ni-input in-modal v-if="!addNewTraineeCompanyStep" @blur="$v.newTrainee.identity.lastname.$touch" caption="Nom"
-          required-field v-model="newTrainee.identity.lastname" :error="$v.newTrainee.identity.lastname.$error" />
-        <ni-input in-modal v-if="!addNewTraineeCompanyStep" v-model.trim="newTrainee.contact.phone"
-          caption="Téléphone" @blur="$v.newTrainee.contact.phone.$touch" :error="$v.newTrainee.contact.phone.$error"
-          :error-message="phoneNbrError($v.newTrainee)" :last="isIntraCourse" />
-        <ni-select v-if="!isIntraCourse" in-modal v-model.trim="newTrainee.company" required-field caption="Structure"
-          @blur="$v.newTrainee.company.$touch" :error="$v.newTrainee.company.$error" :options="companyOptions"
-          :last="!isIntraCourse" />
-      </template>
-      <template slot="footer">
-        <q-btn v-if="firstStep" no-caps class="full-width modal-btn" label="Suivant" color="primary"
-          :loading="additionModalLoading" icon-right="add" @click="nextStepAdditionModal" />
-        <q-btn v-else no-caps class="full-width modal-btn" color="primary" label="Ajouter à la formation"
-          :loading="additionModalLoading" icon-right="add" @click="addTrainee" />
-      </template>
-    </ni-modal>
+    <trainee-creation-modal v-model="traineeCreationModal" :new-trainee="newTrainee" :company-options="companyOptions"
+      :first-step="firstStep" :add-new-trainee-company-step="addNewTraineeCompanyStep" :is-intra-course="isIntraCourse"
+      :validations="$v.newTrainee" :loading="traineeCreationModalLoading" @hide="resetAddTraineeForm"
```

Comment:
dans `resetAddTraineeForm` il faut enlever le `this.traineeCreationModal = false;`